source/base/ByteBuffer.ooc

```diff
@@ -50,11 +54,14 @@ ByteBuffer: class {
 	copyTo: func (other: This, start: Int, destination: Int, length: Int) {
 		memcpy(other pointer + destination, this pointer + start, length)
 	}
-	new: static func ~size (size: Int) -> This { _RecyclableByteBuffer new(size) }
+	new: static func ~size (size: Int, allocator := Allocator defaultAllocator()) -> This { _RecyclableByteBuffer new(size, allocator) }
 	new: static func ~recover (pointer: Byte*, size: Int, recover: Func (This) -> Bool) -> This {
 		_RecoverableByteBuffer new(pointer, size, recover)
 	}
-	free: static func ~all { _RecyclableByteBuffer _free~all() }
+	free: static func ~all {
+		_RecyclableByteBuffer _free~all()
+		Allocator free~all()
```

Comment:
I think I've tried that and the order was messed up. Maybe we could use a `priority` integer value instead of `bool` to control the cleanup order.
But ok, for now I can put it after the line 67.